plugins/org.eclipse.osee.framework.ui.skynet/src/org/eclipse/osee/framework/ui/skynet/BaseImage.java

```diff
@@ -76,7 +81,35 @@ public static KeyedImage getBaseImageEnum(Artifact artifact) {
       try {
          String extension = artifact.getSoleAttributeValue(CoreAttributeTypes.Extension, "");
          if (Strings.isValid(extension)) {
-            return new ProgramImage(extension);
+            // Case 1: Handle artifact with an "xml" extension
```

Comment:
This javadoc "Case1...Sub-case..." is AWESOME!!  It helps describe what's going on without having to read the complex code.

Reply:
Thanks! :)